dev-docs/RFCs/v5.0/purejs-standalone-rfc.md

```diff
@@ -0,0 +1,148 @@
+- Introducing casual developers to prototyping with deck.gl
+- Sharing code snippets and proof-of-concepts, e.g. [#1171](https://github.com/uber/deck.gl/pull/1171)
+- Reporting issues, e.g. [#1071](https://github.com/uber/deck.gl/issues/1071)
+
+
+## Proposed API
+
+The standalone version of deck.gl exposes two global objects: `DeckGL` and `LumaGL`. `DeckGL` contains all exports from `deck.gl/core` and `deck.gl/core-layers`. `LumaGL` contains all exports of `luma.gl`.
+
+`DeckGL` is also a wrapper class that manages shared states between a map, a deck.gl canvas and a viewport controller.
+
+### Constructor
+
+Create a new deck.gl visualization.
+
+```
+const deckgl = new DeckGL(props);
+```
+
+### Properties
+
+All properties of `DeckGLJS`, except `canvas`, `width` and `height` (controlled by the `container` prop).
+
+All properties of the viewport controller (see the `controller` prop).
+
+##### `container` (DOM Element)
+
+Container to append the deck.gl canvas to. The deck.gl canvas is automatically resized to fill the container.
+
+Default `document.body`.
+
+##### `map` (Object)
+
+The base map instance.
+
+By default, if the global variable `mapboxgl` is found, attempt to create a mapbox-gl map. Using Mapbox requires the mapbox-gl library be included:
+
+```
+<script src='https://api.tiles.mapbox.com/mapbox-gl-js/v0.44.1/mapbox-gl.js'></script>
+<link href='https://api.tiles.mapbox.com/mapbox-gl-js/v0.44.1/mapbox-gl.css' rel='stylesheet' />
+```
+
+And an access token be set:
+
+```
+mapboxgl.accessToken = '<mapbox_access_token>';
+```
+
+To use a custom map, supply a `map` object that contains two methods: `setProps(props)` and `finalize()`.
+
+Set `map` to `null` to disable base map.
+
+##### `mapStyle` (String | Object)
+
+Mapbox map style.
+
+
+##### `controller` (Object)
+
+The viewport controller instance.
+
+By default, attempt to create a controller that works with the deck.gl viewport (`MapControllerJS` for `WebMercatorViewport`, or `OrbitControllerJS` for `OrbitViewport`).
+
+To use a custom controller, supply a `controller` object that contains two methods: `setProps(props)` and `finalize()`. The controller may invoke `props.onViewportChange` to update the viewport.
+
+Set `controller` to `null` to disable interaction.
+
+
+### Methods
+
+##### `pickObject`
+
+Same as `DeckGLJS.pickObject`.
+
+##### `pickObjects`
+
+Same as `DeckGLJS.pickObjects`.
+
+##### `setProps`
+
+Update props.
+
+##### `getMapboxMap`
+
+Returns the mapbox Map instance.
+
+##### `finalize`
+
+Release all resources.
+
+
+## Example code
+
+Include the following snippet in HTML:
+```
+<script src="https://some.cdn.domain/deck.gl-5.1.0.min.js"></script>
+```
+
+Hello world example:
+```
+const data = [
+  { pickup: [-122.42, 37.8], dropoff: [-122.48, 37.76] },
+  { pickup: [-122.43, 37.8], dropoff: [-122.42, 37.75] }
+];
+
+const deckgl = new DeckGL({
```

Comment:
I think the DeckGL is changed to be Deck on master

Reply:
As the title says, this is a pure JS version of deck.gl, published as a new npm module.